src/sage/algebras/quatalg/quaternion_algebra_element.pyx

```diff
@@ -703,8 +703,8 @@ cdef class QuaternionAlgebraElement_abstract(AlgebraElement):
             True
         """
         if base_map is None:
-            base_map = lambda v: v
-        return sum(base_map(c)*g for c,g in zip(self, [1] + list(im_gens)))
+            return sum(c * g for c, g in zip(self, [1] + list(im_gens)))
+        return sum(base_map(c) * g for c, g in zip(self, [1] + list(im_gens)))
```

Comment:
This change makes the code harder to read - as this way one has to manually find the difference in handling of this special case. The original version is self-documenting. It would be fine if accompanied by a comment, like  `# just drop base_map() call`

Reply:
I have undone this change.